.github/workflows/_classic-buildpack-prepare-release.yml

```
@@ -74,7 +74,13 @@ jobs:
 
       - name: Generate list of unreleased commits
         id: unreleased-commits
-        run: echo "commits=$(git log --topo-order --reverse --format='- %s' v${{ steps.existing-version.outputs.version }}...main)" >> "${GITHUB_OUTPUT}"
+        # See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+        run: |
+          {
+            echo 'commits<<COMMITS_LIST_END'
+            git log --topo-order --reverse --format='- %s' v${{ steps.existing-version.outputs.version }}...main
+            echo COMMITS_LIST_END
+          } >> "${GITHUB_OUTPUT}"
```

if you want to make it more secure you can use a randomly generated delimiter:

```suggestion
          delimiter="$(openssl rand -hex 8)"
          {
            echo "commits<<${delimiter}"
            git log --topo-order --reverse --format='- %s' v${{ steps.existing-version.outputs.version }}...main
            echo "${delimiter}"
          } >> $GITHUB_OUTPUT
```

---

Yeah I noticed that approach in the Node.js repo, however decided against it in favour of just using a very specific string for simplicity (the `libcnb.rs` usage also uses the simpler approach). I think in practice the main thing to avoid is using a generic string like `EOF` which would have a higher chance of collision.